service/src/test/java/bio/terra/catalog/common/RequestContextCopierTest.java

```diff
@@ -0,0 +1,27 @@
+package bio.terra.catalog.common;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+public class RequestContextCopierTest {
+  @Test
+  void setsRequestContextInsideThread() {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.setAttribute("attribute", "value");
+    RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+    RequestContextCopier.parallelWithRequest(Stream.of(1, 2))
+        .forEach(
```

Comment:
I was wondering how you were going to test this!

Can you add a negative test too?

Also, using `forEach(..assertEquals())` could result in a false positive since it will succeed even if `forEach()` never runs. One way around this would be to use a terminal operation like `sum()` and then assert that the terminal state of the stream is correct as well, or have the `map()` consume the request scope state and use it for its operation.